### ISSUE TITLE
function nextEvent(...) in Meeting.sol

### ISSUE THREAD

**Issue by easonchai** [easonchai/hackmoney-project#6]
Should have a cooldown period as I can transfer to next period before anyone withdraws their stake

**Comment by easonchai** [comment]
Also change the address.transfer to a selfdestruct function to save gas

**Comment by BnWng** [comment]
> Should have a cooldown period as I can transfer to next period before anyone withdraws their stake

I will implement the cooldown period. However, ...

I've realised at the moment the organiser cannot deploy multiple Meetings in advance as we wanted. Since they would either: (a) be transferring the stake prematurely or (b) prevented from doing this by the cooldown period. Will think about this some more.

**Comment by PatrickAlphaVantage** [comment]
Hey guys! I just saw that you're looking to do some timing and I wanted to introduce you to a decentralized timing tech, it's called a Chainlink alarm. I think it may help with scheduling meetings. If I understand your code correctly, you could schedule the date for a meeting ahead of time. 

https://docs.chain.link/docs/chainlink-alarm-clock

**Comment by BnWng** [comment]
Hi @PatrickAlphaVantage , that's good to know, thanks! You're right, some time triggered functions would be useful. Didn't know chainlink had this product. 

However, from the linked document: "1 LINK is required per-request" - so that would be 1 LINK per triggered function call? I think this is would be quite expensive for our meetups usecase.

**Comment by PatrickAlphaVantage** [comment]
Yeah it's one of their best kept secrets haha, decentralized timing oracles. 

Correct, that's just the cost of the testnet implementation, (they keep it at 1LINK for simplicity, you can [get testnet LINK here](https://ropsten.chain.link/) though) its currently only on ropsten, rinkeby, and kovan. On mainnet it would be much less expensive. Average jobs are about 0.1 LINK or less.
